src/main/java/org/cbioportal/legacy/persistence/mybatis/StructuralVariantMyBatisRepository.java

```diff
@@ -52,20 +49,8 @@ public List<StructuralVariant> fetchStructuralVariants(
     if (molecularProfileIds == null || molecularProfileIds.isEmpty()) {
       return new ArrayList<>();
     }
-    return molecularProfileCaseIdentifierUtil
-        .getGroupedCasesByMolecularProfileId(molecularProfileIds, sampleIds)
-        .entrySet()
-        .stream()
-        .flatMap(
-            entry ->
-                structuralVariantMapper
-                    .fetchStructuralVariants(
-                        asList(entry.getKey()),
-                        new ArrayList<>(entry.getValue()),
-                        entrezGeneIds,
-                        structuralVariantQueries)
-                    .stream())
-        .collect(Collectors.toList());
+    return structuralVariantMapper.fetchStructuralVariants(
```

Comment:
this change was made because fetching all profiles at once is much more performant.  it is always true that sampleId collection and molecular profile collection will be of equal length